src/models/user.ts

```diff
@@ -1,4 +1,5 @@
 export interface User {
+  id?: string;
```

Comment:
Możesz chyba zostawić jako wymagane pole a tam gdzie potrzebujesz to TS ma coś takiego jak `Omit`